pkg/operator/controller/ingress/status_test.go

```diff
@@ -203,46 +203,50 @@ func TestComputePodsScheduledCondition(t *testing.T) {
 		},
 	}
 	tests := []struct {
-		name       string
-		deployment *appsv1.Deployment
-		pods       []corev1.Pod
-		expect     operatorv1.ConditionStatus
+		name        string
+		deployment  *appsv1.Deployment
```

Comment:
I always test to ensure functions can handle nil pointers. Do you agree with testing this edge case?

Reply:
I'm ambivalent in this case and in the general case.  Adding nil checks everywhere can make code safer (less likely to panic), but it can also obscure bugs (maybe it's a bug in the caller if the argument is nil).  In this case, the caller of `checkPodsScheduledForDeployment` itself dereferences `deployment`, so it cannot be nil.  For short helper functions like this that are used in only one or two places, I would be fine with saying something like, "The caller must ensure that the deployment argument is not nil."  Would a comment in the godoc for `checkPodsScheduledForDeployment` suffice, or do you still prefer an explicit nil check in `checkPodsScheduledForDeployment`?

Reply:
Actually, `checkPodsScheduledForDeployment` returns an error value, so I'll go ahead and add a check that returns an error if `deployment` is nil.

Reply:
Added.

Reply:
So I was a bit of a NIL Pointer Check absolutist until you mentioned `it can also obscure bugs`. So I agree, it's not that cut-and-dry whether every function needs a NIL pointer check. The GO community seems torn on it too based on my web searches.

So I think I don't mind either way.